src/index.js

```diff
@@ -1,27 +1,39 @@
 import babel from 'babel-core';
-import babiliPreset from 'babel-preset-babili';
+import babelPresetMinify from 'babel-preset-minify';
 import { SourceMapSource, RawSource } from 'webpack-sources';
 
-export default class BabiliPlugin {
-  constructor(babiliOpts = {}, options = {}) {
-    this.babiliOpts = babiliOpts;
-    this.options = options;
-  }
-
-  apply(compiler) {
-    const { babiliOpts, options } = this;
+// alias for undefined / void 0 :(
+// because of these eslint rules in webpack-defaults
+// 1. no-void
+// 2. no-undefined
+let undef;
 
-    const jsregex = options.test || /\.js($|\?)/i;
-    const commentsRegex = typeof options.comments === 'undefined' ? /@preserve|@licen(s|c)e/ : options.comments;
+function getDefault(actualValue, defaultValue) {
+  return actualValue !== undef ? actualValue : defaultValue;
+}
 
-    const useSourceMap = typeof options.sourceMap === 'undefined' ? !!compiler.options.devtool : options.sourceMap;
+export default class BabelMinifyPlugin {
+  constructor(minifierOpts = {}, pluginOpts = {}) {
+    this.options = {
+      parserOpts: pluginOpts.parserOpts || {},
+      minifyPreset: pluginOpts.minifyPreset || babelPresetMinify,
+      minifierOpts,
+      babel: pluginOpts.babel || babel,
+      comments: getDefault(pluginOpts.comments, /^\**!|@preserve|@license|@cc_on/),
+      // compiler.options.devtool overrides options.sourceMap if NOT set
+      // so we set it to undefined/void 0 as the default value
+      sourceMap: getDefault(pluginOpts.sourceMap, undef),
+      jsregex: pluginOpts.jsregex || /\.js($|\?)/i,
```

Comment:
`test` is better imho, since webpack uses this name all over the place for filtering files :)